custom_components/frigate/camera.py

```diff
@@ -139,20 +141,25 @@ def __init__(
             f"{frigate_config['mqtt']['topic_prefix']}" f"/{self._cam_name}/motion/set"
         )
 
-        streaming_template = config_entry.options.get(
-            CONF_RTMP_URL_TEMPLATE, ""
-        ).strip()
-
-        if streaming_template:
-            # Can't use homeassistant.helpers.template as it requires hass which
-            # is not available in the constructor, so use direct jinja2
-            # template instead. This means templates cannot access HomeAssistant
-            # state, but rather only the camera config.
-            self._stream_source = Template(streaming_template).render(
-                **self._camera_config
-            )
+        if self._camera_config.get("restream", {}).get("enabled"):
+            self._stream_source = f"rtsp://{URL(self._url).host}:8554/{self._cam_name}"
```

Comment:
I think the CONF_RTMP_URL_TEMPLATE style approach should apply here too.

Can one camera have restream enabled, but a different camera not? Based on the above, I think so, so my suggestion would be to add a new config option CONF_RTSP_URL_TEMPLATE option that is expanded exactly the same as RTMP below. This is much more extensible since it means we don't make hard-coded assumptions about what host or port number the RTSP feed coming from.